ZCRMiOS/Classes/APIRequest.swift

```diff
@@ -65,17 +66,52 @@ internal class APIRequest
 		self.headers = handler.getRequestHeaders()
 		self.requestBody = handler.getRequestBody()
 		self.isOAuth = handler.getRequestType()
+        self.jsonRootKey = handler.getJSONRootKey()
 	}
 	
-    private func authenticateRequest()
+    private func authenticateRequest( completion : @escaping( Error? ) -> () )
     {
+        self.addHeader( headerName : "User-Agent", headerVal : "Zoho CRM iOS SDK" )
         if( APPTYPE == "ZCRM" )
         {
-            self.addHeader( headerName : "Authorization", headerVal : "Zoho-oauthtoken \( ZCRMLoginHandler().getOauth2Token() )" )
+            ZCRMLoginHandler().getOauth2Token { ( token, error ) in
+                if let err = error
+                {
+                    completion( err )
+                }
+                if let oAuthtoken = token, token.notNilandEmpty
+                {
+                    self.addHeader( headerName : "Authorization", headerVal : "Zoho-oauthtoken \( oAuthtoken )" )
+                    completion( nil )
+                }
+                else
+                {
+                    print( "oAuthtoken is nil." )
+                    completion( UnexpectedError.ResponseNil( "oauthtoken is empty" ) )
```

Comment:
Can we change UnexpectedError to ZCRMSDKError and rename the existing ZCRMSDKError as ZCRMAppError or ZCRMClientError or simply ZCRMError?